### PR TITLE
App : Errors - New - Create an handler for unexpected errors.

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

**Review by matipojo — DISMISSED** [review]

File: core/app/assets/js/organisms/error-boundary.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+			hasError: null,
+		};
+	}
+
+	static getDerivedStateFromError() {
+		return { hasError: true };
+	}
+
+	goBack() {
+		// If the app was open inside iframe it just close it,
```
Comment (kobizz) [inline core/app/assets/js/organisms/error-boundary.js:33]:
```suggestion
		// If the app was open inside iframe it just closes it,
```

File: core/app/assets/js/organisms/error-boundary.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		// if not it redirect back to the last location.
```
Comment (kobizz) [inline core/app/assets/js/organisms/error-boundary.js:34]:
```suggestion
		// if not it redirects back to the last location.
```

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/app/assets/js/organisms/error-boundary.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		children: PropTypes.any,
+		title: PropTypes.string,
+		text: PropTypes.string,
+		learnMoreUrl: PropTypes.string,
+	};
+
+	static defaultProps = {
+		title: __( 'App could not be loaded', 'elementor' ),
+		text: __( 'We’re sorry, but something went wrong. Click on ‘Learn more’ and follow each of the steps to quickly solve it.', 'elementor' ),
+		learnMoreUrl: 'https://go.elementor.com/app-general-load-issue/',
```
Comment (kobizz) [inline core/app/assets/js/organisms/error-boundary.js:17]: It's better that these values won't be provided as default values, but will be given from the implementation

**Review by kobizz — CHANGES_REQUESTED** [review]

File: core/app/assets/js/organisms/error-boundary.js (hunk 3 above)
Comment (matipojo) [reply core/app/assets/js/organisms/error-boundary.js:17]: It must have default values because it's a default generic error.

**Review by kobizz — DISMISSED** [review]

**Review by matipojo — DISMISSED** [review]

**Review by matipojo — APPROVED** [review]

**Pull request merged**